src/main/java/org/apache/sling/commons/threads/impl/DefaultThreadPoolManager.java

```diff
@@ -404,23 +426,56 @@ public ThreadPoolExecutor getExecutor() {
             return null;
         }
 
-        protected void unregisterMBean() {
+        protected void unregisterMBeanAndMetrics() {
             if ( this.mbeanRegistration != null ) {
                 this.mbeanRegistration.unregister();
                 this.mbeanRegistration = null;
             }
+            unregisterGauges();
         }
 
-        protected void registerMBean() {
+        protected void registerMBeanAndMetrics() {
             try {
                 final Dictionary<String, String> mbeanProps = new Hashtable<String, String>();
                 mbeanProps.put("jmx.objectname", "org.apache.sling:type=threads,service=ThreadPool,name=" + this.name);
 
                 final ThreadPoolMBeanImpl mbean = new ThreadPoolMBeanImpl(this);
                 this.mbeanRegistration = bundleContext.registerService(ThreadPoolMBean.class.getName(), mbean, mbeanProps);
+
+                // not exposing MaxThreadAge as bean.getMaxThreadAge() is deprecated due
+                // to SLING-6261 and always returns -1
+                String[] ignoredAttributes = { "MaxThreadAge" };
+
+                for(Map.Entry<String, Gauge<?>> nameToGauge : ThreadPoolMetricsGauges.create(mbean, mbean.getMBeanInfo(),
+                        Arrays.asList(ignoredAttributes)).entrySet()) {
+                    addGauge(nameToGauge.getKey(), nameToGauge.getValue());
+                }
             } catch (Throwable t) {
                 logger.warn("Unable to register Thread Pool MBean", t);
             }
         }
+
+        private void addGauge(String name, Gauge<?> gauge) {
+            final String fullName = getMetricFullName(name);
+            final Dictionary<String, Object> props = new Hashtable<>();
+            props.put(Gauge.NAME, fullName);
+            props.put(Constants.SERVICE_DESCRIPTION, "Apache Sling Metrics Gauge - " + fullName);
+            props.put(Constants.SERVICE_VENDOR, "The Apache Software Foundation");
+
+            try {
+                gaugeRegistrations.add(bundleContext.registerService(Gauge.class, gauge, props));
+            } catch (Throwable t) {
```

Comment:
Do we need `Throwable` here or can we catch more specific exceptions?

Reply:
it mimics pre-existing line 453 where ThreadPoolMBean is registered as a service in BundleContext - it could be changed of course to IllegalStateException only (as declared) in context's javadocs - but question if it should be also changed for the MBean registration?

Reply:
@rombert on the second thought I've changed both places to use as fine grained exceptions as possible (looking both at checked exceptions and the declaration in API's javadocs)